app-deployment-manager/pkg/fleet/configgen.go

```diff
@@ -794,3 +821,42 @@ func WriteExtraValues(basedir string, filename string, e *ExtraValues) error {
 	}
 	return utils.WriteFile(basedir, filename, data)
 }
+
+func getRegistryProjectName(d *v1beta1.Deployment, kc client.Client) (string, error) {
+	projectID := d.Labels[string(v1beta1.AppOrchActiveProjectID)]
+	if projectID == "" {
+		return "", fmt.Errorf("project-id not found in deployment labels")
+	}
+	log.Infof("Look for project-id %s in Nexus", projectID)
+	cfg, err := rest.InClusterConfig()
```

Comment:
@adibrastegarnia @badhrinathpa I've got the nexusClient here as I have seen in other places, but I cannot see how to add this in to a Unit Test. Is there some reference I could use?

As an alternative I could just use `kc` client to get the Nexus CRD `RuntimeProject` and just get the values from this? Which do you think is better.